crates/walrus-service/src/node/events/event_processor.rs

```diff
@@ -706,6 +718,9 @@ impl EventProcessor {
                 .stores
                 .checkpoint_store
                 .insert(&(), verified_checkpoint.serializable_ref())?;
+
+            // Also update the cache with the bootstrap checkpoint sequence number.
+            event_processor.update_checkpoint_cache(*verified_checkpoint.sequence_number());
```

Comment:
We also need to update the checkpoint in the `catchup_using_event_blobs` path where we are updating events using event blobs.